scripts/update_capacity_configuration.py

```diff
@@ -0,0 +1,220 @@
+from copy import deepcopy
+from logging import INFO, basicConfig, getLogger
+from typing import Any
+
+from electricitymap.contrib.config import CONFIG_DIR, ZONE_PARENT
+from electricitymap.contrib.config.constants import PRODUCTION_MODES, STORAGE_MODES
+from electricitymap.contrib.config.reading import read_zones_config
+from electricitymap.contrib.lib.types import ZoneKey
+from scripts.utils import write_zone_config
+
+logger = getLogger(__name__)
+basicConfig(level=INFO)
+ZONES_CONFIG = read_zones_config(CONFIG_DIR)
+CAPACITY_MODES = PRODUCTION_MODES + [f"{mode} storage" for mode in STORAGE_MODES]
+
+
+def sort_config_keys(config: dict[str, Any]) -> dict[str, Any]:
+    """Sort the keys of the config dict"""
+    return {k: config[k] for k in sorted(config)}
+
+
+def update_zone_capacity_config(zone_key: ZoneKey, data: dict) -> None:
+    """Update the capacity config for a zone"""
+    if zone_key not in ZONES_CONFIG:
+        raise ValueError(f"Zone {zone_key} does not exist in the zones config")
+
+    _new_zone_config = deepcopy(ZONES_CONFIG[zone_key])
+    if "capacity" in _new_zone_config:
+        capacity = _new_zone_config["capacity"]
+
+        if all(isinstance(capacity[m], float | int) for m in capacity.keys()):
+            capacity = data
+        else:
+            capacity = get_zone_capacity_config(capacity, data)
+    else:
+        capacity = data
+
+    _new_zone_config["capacity"] = capacity
+
+    # sort keys
+    _new_zone_config["capacity"] = sort_config_keys(_new_zone_config["capacity"])
+
+    write_zone_config(zone_key, _new_zone_config)
+
+
+def get_zone_capacity_config(
```

Comment:
There seems to be a mismatch between the function name and the description, the name is get, while the description says that we update.